pybamm/models/submodels/transport_efficiency/general_transport_efficiency.py

```diff
@@ -0,0 +1,83 @@
+#
+# Class for Bruggemantransport_efficiency
+#
+import pybamm
+from .base_transport_efficiency import BaseModel
+
+
+class GeneralTransportEfficiency(BaseModel):
```

Comment:
Do we anticipate transport efficiency models that would not be into `general_transport_efficiency.py`? In that case, it may be better to get rid of the base class and just have a `transport_efficiency.py` file. Alternatively, we could split `general_transport_efficiency.py` into subclasses (one for each model). Not sure what's best practice in these situations honestly, so happy to be convinced either way.

Reply:
One advantage of the latter is that when checking the submodel components it will be a lot more explicit.

Reply:
For kinetics, each different option is its own class that just implements `_get_kinetics`